File: app/src/main/java/com/project/major/alumniapp/utils/ConnectionHelper.java
/******************************************************************************
 * Copyright (c) 2020.                                                        *
 * Christin B Koshy.                                                          *
 * 3                                                                          *
 ******************************************************************************/

package com.project.major.alumniapp.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.util.Objects;

public class ConnectionHelper {
    public static long lastNoConnectionTs = -1;
    public static boolean isOnline = true;

    public static boolean isConnected(Context context){
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = Objects.requireNonNull(cm).getActiveNetworkInfo();
        return activeNetwork != null && activeNetwork.isConnected();
    }

    public static boolean isConnectedOrConnecting(Context context) {
        ConnectivityManager cm =(ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = Objects.requireNonNull(cm).getActiveNetworkInfo();

        return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/project/major/alumniapp/utils/ConnectionHelper.kt
/******************************************************************************
 * Copyright (c) 2020.                                                        *
 * Christin B Koshy.                                                          *
 * 3                                                                          *
 */
@file:Suppress("DEPRECATION")

package com.project.major.alumniapp.utils

import android.content.Context
import android.net.ConnectivityManager
import java.util.*

object ConnectionHelper {
    var lastNoConnectionTs: Long = -1
    var isOnline = true
    fun isConnected(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = Objects.requireNonNull(cm).activeNetworkInfo
        return activeNetwork != null && activeNetwork.isConnected
    }

    fun isConnectedOrConnecting(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = Objects.requireNonNull(cm).activeNetworkInfo
        return activeNetwork != null && activeNetwork.isConnectedOrConnecting
    }
}